website/docs/terms/subquery.md

```diff
@@ -0,0 +1,220 @@
+---
+id: subquery
+title: Subquery in SQL
+displayText: subquery
+hoverSnippet: A subquery is a query within another query. Subqueries are often used when you need to process data in multiple steps.
+---
+
+A subquery is what the name suggests: a query within another query. The true inception of SQL. Subqueries are often used when you need to process data in several steps. For the majority of subqueries you’ll see in actual practice, the inner query will execute first and pass its result to the outer query it's nested in.
+
+Subqueries are usually contrasted with <Term id="cte">Common Table Expressions (CTEs)</Term> as they have similar use cases. Unlike CTEs, which are usually separate `SELECT` statements within a query, subqueries are usually `SELECT` statements nested within a `JOIN`, `FROM`, or `WHERE` statement in a query.
+
+To be honest, we rarely write subqueries here at dbt Labs since we prefer to use CTEs. We find that CTEs, in general, support better query readability, organization, and debugging. However, subqueries are a foundational concept in SQL and still widely used. We hope you can use this glossary to better understand how to use subqueries and how they differ from CTEs.
+
+## Subquery syntax
+
+While there are technically several types of subqueries, the general syntax to build them is the same. A subquery usually consists of the following:
+
+- Enclosing parentheses
+- A name
+- An actual SELECT statement
+- A main query it is nested in via a FROM, WHERE, or JOIN clause
+
+Let’s take this to an example, using the [sample jaffle_shop dataset](https://github.com/dbt-labs/jaffle_shop).
+
+```sql
+select customer_id, count(order_id) as cnt_orders
+  from (
+        select * from {{ ref('orders') }}
+       ) all_orders
+group by 1
+```
+
+Given the elements of subqueries laid out in the beginning, let’s break down this example into its respective parts.
+
+| Subquery elements | Example |
+|---|---|
+| Enclosing parentheses | :white_check_mark: |
+| Subquery name | `all_orders` |
+| `SELECT` statement | `select * from {{ ref('orders') }}` | 
+| Main query it is nested in | `select customer_id, count(order_id) as cnt_orders from all_orders group by 1` | 
+
+When this query is actually executed, it will start by running the innermost query first. In this case, it would run `select * from {{ ref(‘orders’) }}` first. Then, it would pass those results to the outer query, which is where you grab the count of orders by `customer_id`.
```

Comment:
```suggestion
When this query is actually executed, it will start by running the innermost query first. In this case, it would run `select * from {{ ref('orders') }}` first. Then, it would pass those results to the outer query, which is where you grab the count of orders by `customer_id`.
```